.github/workflows/test.yml

```diff
@@ -51,8 +51,31 @@ jobs:
             pip install tox tox-gh-actions
         - name: Run integration tests
           run: tox -eintegration
+    changedfiles:
```

Comment:
I wonder if this would work better as a custom action (same code, but called differently, as shown here - https://docs.github.com/en/actions/creating-actions/creating-a-composite-run-steps-action)?  I am still not sure if the end result here will achieve what we want, though perhaps it will.